src/GregClientTests/GregClientTests.cs

```diff
@@ -34,7 +37,119 @@ public void GetPackageVersionHeaderTest()
+            Console.WriteLine(request.RequestBody.AsJson());
+        }
+        
+        [Test]
+        public void UploadDynamoPackageWithHostDependencyTest()
+        {
+            var keywords = new List<string>() { "Civil" };
+            var nv = new PackageVersionUploadRequestBody("Third .NET Package", "2.1.0", "", keywords, "contents", "dynamo", "0.1.0", "metadata", "group",
+                            new List<PackageDependency>() { new PackageDependency("Ram", "0.1.0"), new PackageDependency("Ian", "0.1.0") }, "", "",
+                            false, new List<String>(), new List<String>() { "Civil3D" }, "Dynamo Team", "2021");
+
+            var files = new List<string>() { "../test/pedro.dyf", "../test/RootNode.dyf" };
+            var request = new PackageVersionUpload(nv, files);
+            Assert.That(request.RequestBody.AsJson().Equals("{\"file_hash\":null,\"name\":\"Third .NET Package\",\"version\":\"2.1.0\",\"description\":\"\",\"group\":\"group\",\"keywords\":[\"Civil\"],\"dependencies\":[{\"name\":\"Ram\",\"version\":\"0.1.0\"},{\"name\":\"Ian\",\"version\":\"0.1.0\"}],\"host_dependencies\":[\"Civil3D\"],\"contents\":\"contents\",\"engine_version\":\"0.1.0\",\"engine\":\"dynamo\",\"engine_metadata\":\"metadata\",\"site_url\":\"\",\"repository_url\":\"\",\"contains_binaries\":false,\"node_libraries\":[],\"copyright_holder\":\"Dynamo Team\",\"copyright_year\":\"2021\"}"));
+            Console.WriteLine(request.RequestBody.AsJson());
+        }
+        
+        [Test]
+        public void UploadDynamoPackageVersionWithFilesTest()
+        {
+            var keywords = new List<string>() { "neat", "ok" };
+            var nv = new PackageVersionUploadRequestBody("Third .NET Package", "2.1.0", "", keywords, "contents", "dynamo", "0.1.0", "metadata", "group",
+                new List<PackageDependency>() { new PackageDependency("peter", "0.1.0"), new PackageDependency("stephen", "0.1.0") }, "", "", false, new List<String>(), new List<String>(), "", "");
+
+            var files = new List<string>() {Assembly.GetExecutingAssembly().Location };
+
+            var request = new PackageVersionUpload(nv, files);
+
+            var rr = new RestRequest();
+            request.Build(ref rr);
+            Assert.That(request.Files.Count, Is.EqualTo(1));
+            Console.WriteLine(request.RequestBody.AsJson());
+        }
+
+        [Test]
+        public void DownloadDynamoPackageByEngineAndNameTest()
+        {
+            GregClient pmc = new GregClient(null, "http://dynamopackages.com/");
+
+            var nv = new HeaderDownload("dynamo", "MeshToolkit");
+            var pkgResponse = pmc.ExecuteAndDeserializeWithContent<PackageHeader>(nv);
+            Console.WriteLine(JsonSerializer.Serialize(pkgResponse.content)); // the package
```

Comment:
What are you "asserting" for in this test?

Reply:
nothing much, the serialize step will throw if the response can't find the package, but I can add a null check in there if you prefer to be more explicit.